clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java

```diff
@@ -89,10 +89,11 @@ public ResponseHeader toResponseHeader() {
     public static RequestHeader parse(ByteBuffer buffer) {
         short apiKey = -1;
         try {
+            int position = buffer.position();
```

Comment:
Is this specifically for Envelope case?

Reply:
This was a bug. This logic assumes that the buffer is at position 0. Let me add a test case.